src/js/main.js

```diff
@@ -88,67 +69,217 @@ class Timelines extends React.Component {
+		if(this.state.postTweetError) {
+			return "errorPost";
+		}
+		else if(_.isNull(this.state.postTweetMessage) && _.isNull(this.state.postTweetError)) {
+			return "pending";
+		}
+		else {
+			return "successPostTweet";
+		}
+	}
+
+	render() {
+		{this.postTweetResultOutput()}
+		{this.postTweetResultClass()}
+		return e(PostTweetComponent, {onKeyPressButton: this.handleKeyPress,
+									  onChangeButton: this.handleChange,
+									  tweet: this.state.value,
+									  key: 'postTweetComp',
+									  resultClass: this.postTweetResultClass(),
+									  buttonFunc: () => this.handlePostTweet(this.state.value),
+									  resultMessage: this.postTweetResultOutput()});
+	}
+}
+
+class UserTimeline extends React.Component {
+	
+	constructor(props) {
+		super(props);
+		this.state = {
+			userTimeline: null,
+			userTimelineError: null,	
+			value: ''
+		}
+		this.handleGetUserTimeline = this.handleGetUserTimeline.bind(this);
+		this.userTimelineResultOutput = this.userTimelineResultOutput.bind(this);
+		this.userTimelineResultClass = this.userTimelineResultClass.bind(this);
+	}
+
+	handleGetUserTimeline() {
+		getUserTimeline().then(response => response.json())
+					     .then(data => {
+					   	  	this.setState({
+					   			userTimeline: data,
+					   			userTimelineError: false
+					   		})
+					     })
+					     .catch(error => {
+					   		this.setState({
+					   			userTimeline: null,
+					   			userTimelineError: true
+					   		})
+					     });
+	}
+
+	userTimelineResultClass() {
+		if(this.state.userTimelineError) {
+			return "error";	
+		}
+		else if(_.isNull(this.state.userTimeline) && !this.state.userTimelineError) {
+			return "pending";	
+		}
+		else {
+			return "successGetTimeline";			
+		}
+	}
+
+	userTimelineResultOutput() {
+		if(this.state.userTimelineError) {
+			return "There was a problem on the server side, please try again later";
+		}
+		else if(_.isNull(this.state.userTimeline) && !this.state.userTimelineError) {
+			return " ";
+		}
+		else {
+			return renderedTimeline(this.state.userTimeline, false);
+		}
+	}
+
+	componentDidMount() {
+		{this.handleGetUserTimeline()};
+	}
+
+	render() {
+		{this.userTimelineResultClass()};
+		{this.userTimelineResultOutput()};
+		return e(TimelineComponent, {key: 'userTimelineComp', 
 							 	  timelineType: 'User', 
 								  buttonFunc: () => {this.handleGetUserTimeline()}, 
-							 	  resultClass: userTimelineResultClass, 
-							 	  resultOutput:userTimelineResultOutput}, null)]);			
+							 	  resultClass: this.userTimelineResultClass(), 
+							 	  resultOutput: this.userTimelineResultOutput()});	
+	}
+}
+
+class TabsAndTabPages extends React.Component {
+	render() {
+		return e(Tabs, {}, [
+					e(TabList, {},[
+						e(Tab, {label: "homeTimelineTab"}, "Home Timeline"),
+						e(Tab, {label: "userTimelineTab"}, "User Timeline"),
+						e(Tab, {label: "postTweetTab"}, "Post Tweet")]),
+					e(TabPanel, {}, e(HomeTimeline, {})),
```

Comment:
do you need the {} for the components?  (HomeTimeline/UserTimeline/PostTweet)

Reply:
only if there are components inside the created react element but ill remove instances taht are not that case